bindings/gumjs/gumquickprocess.c

```diff
@@ -132,6 +145,8 @@ static const JSCFunctionListEntry gumjs_process_entries[] =
   JS_CFUNC_DEF ("_enumerateMallocRanges", 0,
       gumjs_process_enumerate_malloc_ranges),
   JS_CFUNC_DEF ("setExceptionHandler", 0, gumjs_process_set_exception_handler),
+  JS_CFUNC_DEF ("runOnThreadSync", 0, gumjs_process_run_on_thread_sync),
```

Comment:
I'm wondering if we want to drop the `Sync` variant, and drop the `Async` prefix on the other one, and have it return a `Promise` instead. That way one can `await` it if the result is needed, and still allow the JS MainContext to process events in the meantime. What do you think?

Reply:
Sounds good. I've made a new commit with the changes. Note that I've retained the **sync** and **async** versions in the C API, and then just make use of the **async** version in JS world. Otherwise there is no equivalent for C.

Reply:
Yay, thanks! For the C API, to make it consistent with GIO conventions  what do you think about dropping the _async suffix and only keeping the _sync suffix?

Reply:
I suppose we can drop the "is supported" API, right? (A generic "is Stalker supported" query would be better, if we need it.)

Reply:
Updated as requested.